doc/design/switch_kernel.md

```diff
@@ -0,0 +1,66 @@
+## Background
+Every operator has many kernels. It because there are multiple data types, places, data layout that Fluid supports. We use the `KernelType` to describe kernel types that operators can hold. 
+
+The `KernelType` is as follows.
+
+```
+struct KernelType {
+  Place place_;
+  DataType data_type_;
+  LayoutType layout_;
+};
+```
+
+The `place_` is a descriptor of the device and the computational library, e.g., `MKLPlace`, `CUDAPlace`.
```

Comment:
Do we need to support different kind of kernel with the same KernelType, for example, one CPU kernel is written in native cpp while another one is written with some Blas library.  If we will support, then how to get the right kernel

Reply:
No, we do not.

Reply:
if a user wants to add a custom CPU kernel that is already implemented and registered, how should he register and choose to use it? The data type, place, and layout are all the some.